src/evaluate/hub.py

```diff
@@ -0,0 +1,125 @@
+
+    Args:
+        repo_id (``str``): Model id from https://hf.co/datasets.
+        task_type (``str``): Task id, refer to
+            https://github.com/huggingface/datasets/blob/master/src/datasets/utils/resources/tasks.json for allowed values.
+        dataset_type (``str``): Dataset id from https://hf.co/datasets.
+        dataset_name (``str``): Pretty name for the dataset.
+        metric_type (``str``): Metric id from https://hf.co/metrics.
+        metric_name (``str``): Pretty name for the metric.
+        metric_value (``float``): Computed metric value.
+        task_name (``str``, optional): Pretty name for the task.
+        dataset_config (``str``, optional): Dataset configuration used in datasets.load_dataset().
+            See huggingface/datasets docs for more info: https://huggingface.co/docs/datasets/package_reference/loading_methods#datasets.load_dataset.name
+        dataset_split (``str``, optional): Name of split used for metric computation.
+        dataset_revision (``str``, optional): Git hash for the specific version of the dataset.
+        dataset_args (``dict[str, int]``, optional): Additional arguments passed to datasets.load_dataset().
+        metric_config (``str``, optional): Configuration for the metric, e.g. (e.g. the GLUE metric has a configuration for each subset)
+        metric_args (``dict[str, int]``, optional): Arguments passed during Metric.compute().
+        overwrite (``bool``, optional, defaults to `False`): If set to `True` an existing metric field can be overwritten, otherwise
+             attempting to overwrite any existing fields will cause an error.
+
+    Example:
+
+    ```python
+    >>> push_to_hub(
+    ...     repo_id="huggingface/gpt2-wikitext2",
+    ...     metric_value=0.5
+    ...     metric_type="bleu",
+    ...     metric_name="BLEU",
+    ...     dataset_name="WikiText",
+    ...     dataset_type="wikitext",
+    ...     dataset_split="test",
+    ...     task_type="text-generation",
+    ...     task_name="Text Generation"
+    ... )
+    ```"""
+    tasks = get_allowed_tasks(known_task_ids)
+
+    if task_type not in tasks:
+        raise ValueError(f"Task type not supported. Task has to be one of {tasks}")
+
+    try:
+        dataset_info(dataset_type)
+    except requests.exceptions.HTTPError:
+        logger.warning(f"Dataset {dataset_type} not found on hf.co/datasets")
+
+    try:
+        model_info(repo_id)
+    except requests.exceptions.HTTPError:
+        raise ValueError(f"Model {repo_id} not found on the Hub at hf.co/{repo_id}")
+
+    result = {
+        "task": {
+            "type": task_type,
+        },
+        "dataset": {
+            "type": dataset_type,
+            "name": dataset_name,
+        },
+        "metrics": [
+            {
+                "type": metric_type,
+                "value": metric_value,
+            },
+        ],
+    }
+
+    if dataset_config is not None:
+        result["dataset"]["config"] = dataset_config
```

Comment:
Consider something like this for more conciseness

```py
result["dataset"]["config"] = dataset_config if dataset_config is not None
```

Reply:
I think if I do that I need to add an `else` statement at the end of the line to complete the ternary. I could write it like

```py
if dataset_config is not None: result["dataset"]["config"] = dataset_config
```

but then Black (via `make style`) undoes it 😂

Reply:
Unless, we wanted to ignore rule `E701` for Black, @lvwerra ?

Reply:
Ah then this is good!